OpenSim/Common/Exception.h

```diff
@@ -67,14 +110,31 @@ class OSIMCOMMON_API Exception  : public std::exception {
     std::string _file;
     /** Line number at which the error occurred. */
     int _line;
+    /** Delimiter/separator used to separate different parts of the message.  */
+    static const std::string _msgSeparator;
```

Comment:
The delimiter is a *necessary* part of formatting message. Reviews suggested to move formatting the messages to OpenSim::Exception and I have done so. I don't see a problem with its existence. I will talk to Ajay and get back. I am not sure I understand the problem here exactly.

Regarding the changing access specifiers, I am not sure this PR is meant to modify OpenSim::Exception affecting other code. This PR is meant to be non-invasive. It is true that we need to correct old code but it is probably for another PR. I will talk to Ajay about this too and get back.